sdk/python/feast/job_service.py

```diff
@@ -117,6 +126,22 @@ def StartStreamToOnlineIngestionJob(
         feature_table = self.client.get_feature_table(
             request.table_name, request.project
         )
+
+        if self.client._config.getboolean(CONFIG_JOB_SERVICE_ENABLE_CONTROL_LOOP):
+            # If the control loop is enabled, return existing stream ingestion job id instead of starting a new one
+            params = get_stream_to_online_ingestion_params(
+                self.client, request.project, feature_table, []
+            )
+            job_hash = params.get_job_hash()
```

Comment:
Would it be better to use `__hash__()` on Python objects and to use a class of the job here instead of params?

Reply:
Didn't want to reuse __hash__ which is intended for a different purpose (e.g. deduping in dictionaries & sets). The params class is more general to define its __hash__ for this purpose only.

Reply:
`__hash__` wouldn't really work here anyway since it would change if you restart the pod/process

Reply:
Oh I thought Willem meant overriding & implementing `__hash__` function instead of `get_job_hash`. Using existing `__hash__` directly would just not work as Oleg stated.

Reply:
I meant overriding the __hash__ function so that the unique (and unchanging) properties of the job are used to create a hash. That way we can have `Job1 == Job1` and get away from writing custom getters/setters and storage for hash.

Reply:
I'd def prefer an explicit `get_job_hash()` and make it part of the params interface (like it is rn). If I was reading feast code, overriding `__hash__` in this matter to do this would be surprising to me.